src/constants.zig

```diff
@@ -262,14 +263,15 @@ comptime {
 pub const pipeline_prepare_queue_max: u32 = config.cluster.pipeline_prepare_queue_max;
 
 /// The maximum number of Viewstamped Replication request messages that can be queued at a primary,
-/// waiting to prepare.
-pub const pipeline_request_queue_max: u32 = clients_max -| pipeline_prepare_queue_max;
+/// waiting to prepare. Each client has at most one request in flight, and a primary can send a
+/// pulse or request upgrade.
+pub const pipeline_request_queue_max: u32 = clients_max + 1 -| pipeline_prepare_queue_max;
```

Comment:
The precedence is correct as it is, still it reads better if explicitly stated.

```suggestion
pub const pipeline_request_queue_max: u32 = (clients_max + 1) -| pipeline_prepare_queue_max;
```